File: src/main/java/pl/training/workshop/m8/s04_shadowlimits/step1/Effects.java
package pl.training.workshop.m8.s04_shadowlimits.step1;

import pl.training.workshop.shared.Money;

/** Krok 1: port efektów ubocznych - nowa ścieżka nie sięga już bezpośrednio do infrastruktury. */
public interface Effects {
    void sendMail(String to, String text);

    void charge(String card, Money amount);

    void save(String row);
}

File: typescript/src/workshop/m8/s04_shadowlimits/step1/Effects.ts
import type { Money } from '../../../shared/Money.js';

/** Krok 1: port efektów ubocznych - nowa ścieżka nie sięga już bezpośrednio do infrastruktury. */
export interface Effects {
  sendMail(to: string, text: string): void;

  charge(card: string, amount: Money): void;

  save(row: string): void;
}
